infra/terraform/modules/auth0/main.tf

```diff
@@ -0,0 +1,87 @@
+provider "auth0" {
+  domain        = "${var.auth0_tenant_domain}"
+  client_id     = "${var.auth0_api_client_id}"
+  client_secret = "${var.auth0_api_client_secret}"
+}
+
+resource "auth0_client" "aws" {
+  name        = "test-AWS"
+  description = "SAML provider used for 'Open in AWS'"
+  app_type    = "regular_web"
+  callbacks   = [
+    "https://signin.aws.amazon.com/saml",
+    "https://aws.services.${var.env}.${var.root_domain}/callback",
+  ]
+  allowed_origins = [
+    "https://aws.services.${var.env}.${var.root_domain}",
+  ]
+}
+
+resource "auth0_client" "kubectl-oidc" {
+  name      = "test-kubectl-oidc"
+  app_type  = "regular_web"
+  callbacks = [
+    "http://localhost:3000/callback",
+    "https://cpanel-master.services.${var.env}.${var.root_domain}/callback",
+  ]
+  allowed_origins = [
+    "http://localhost:3000",
+    "https://cpanel-master.services.${var.env}.${var.root_domain}",
+  ]
+  allowed_logout_urls = [
+    "http:/localhost:3000",
+    "https://cpanel-master.services.${var.env}.${var.root_domain}",
+  ]
+}
+
+resource "auth0_connection" "github" {
+  name = "test-github"
+  strategy = "github"
+  options  = {
+    configuration = {
+      client_id     = "${var.github_oauth_client_id}"
+      client_secret = "${var.github_oauth_client_secret}"
+      profile       = true
+      email         = true
+      read_user     = true
+      read_org      = true
+      scope         = "user:email,read:user,read:org"
+    }
+  }
+  enabled_clients = [
+    "${auth0_client.aws.id}",
+    "${auth0_client.kubectl-oidc.id}",
+  ]
+}
+
+locals {
+  auth0_rules_config = "${map(
```

Comment:
Any reason you can't use a literal map here?

```
auth0_rules_config = {
  foo = "bar"
}
```

Reply:
Ah - I was stuck on trying to make `auth0_rules_config` a variable and not being able to have interpolated values. When I switched to `locals`, I didn't realise I could use a literal map again :facepalm: